third_party/libpg_query/include/parser/parser.hpp

```diff
@@ -29,7 +29,7 @@ typedef enum PGBackslashQuoteType {
 /* Primary entry point for the raw parsing functions */
 PGList *raw_parser(const char *str);
 
-bool is_keyword(const char *str);
+bool is_keyword(const char *str, const std::vector<duckdb_libpgquery::PGKeywordCategory> &type_filter = {});
```

Comment:
Can we replace this with returning the keyword category instead, and adding a `PG_KEYWORD_NONE` category? Then we don't need to pass in this vector and do the comparisons here